samples/vm_perf_example.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+ Written by Lance Hasson
+ Github: https://github.com/JLHasson
+
+ Script to report all available realtime performance metrics from a
+ virtual machine. Based on a Java example available in the VIM API 6.0
+ documentationavailable online at:
+ https://pubs.vmware.com/vsphere-60/index.jsp?topic=%2Fcom.vmware.wssdk.pg.
+ doc%2FPG_Performance.18.4.html&path=7_1_0_1_15_2_4
+"""
+
+from pyVmomi import vim
+from tools import cli
+from pyVim.connect import SmartConnectNoSSL, Disconnect
+import atexit
+import getpass
+
+
+def main():
+
+    args = cli.get_args()
+
+    # Connect to the host without SSL signing
+    try:
+        si = SmartConnectNoSSL(
+            host=args.host,
+            user=args.user,
+            pwd=args.password,
+            port=int(args.port))
+        atexit.register(Disconnect, si)
+
+    except IOError as e:
+        pass
+
+    if not si:
+        raise SystemExit("Unable to connect to host with supplied info.")
+
+    content = si.RetrieveContent()
+    perfManager = content.perfManager
+
+    # create a mapping from performance stats to their counterIDs
+    # counterInfo: [performance stat => counterId]
+    # performance stat example: cpu.usagemhz.LATEST
+    # counterId example: 6
+    counterInfo = {}
+    for c in perfManager.perfCounter:
+        prefix = c.groupInfo.key
+        fullName = c.groupInfo.key + "." + c.nameInfo.key + "." + c.rollupType
+        counterInfo[fullName] = c.key
+
+    # create a list of vim.VirtualMachine objects so
+    # that we can query them for statistics
+    container = content.rootFolder
+    viewType = [vim.VirtualMachine]
+    recursive = True
+
+    containerView = content.viewManager.CreateContainerView(container,
+                                                            viewType,
+                                                            recursive)
+    children = containerView.view
+
+    # Loop through all the VMs
+    for child in children:
+        # Get all available metric IDs for this VM
+        counterIDs = [m.counterId for m in
```

Comment:
@tianhao64 Can you tell me what counterIDs holds when you run the script on your environment?